templates/default/customer/customeradd.html

```diff
@@ -153,15 +153,11 @@ <H1>{$layout.pagetitle}</H1>
 						<IMG SRC="img/folder.gif" alt="">
 					</TD>
 					<TD>
-						<label>
+						<label {tip text="Select legal personality type" trigger="type"}>
 							<input type="radio" name="customeradd[type]" value="{$smarty.const.CTYPES_PRIVATE}" id="private"
-								{tip text="Select legal personality type" trigger="type"}
-								{if $customeradd.type == $smarty.const.CTYPES_PRIVATE} checked{/if}>
+								{if isset($customeradd.type) && $customeradd.type == $smarty.const.CTYPES_PRIVATE} checked{/if} required>
 							{$_CTYPES[$smarty.const.CTYPES_PRIVATE]}
```

Comment:
Ozaczenie typu klienta "prywatny" jako required? Bardzo dziwne...

Reply:
Proszę o opinię @awbnet ;-)

Reply:
U nas rozwiąże to problem omyłkowego nie zaznaczania  "firmy" podczas dodawania klienta. Szczególnie było to widać weryfikując raport TransGus.

Reply:
W UI wygląda to tak: ![radiotype](https://user-images.githubusercontent.com/17087236/106469957-40d24880-64a0-11eb-91a3-bc2d5356b4b1.png)

Reply:
Mam mieszane odczucia trochę, z przewagą na głos na nie. @interduo jak wiesz nie jestem fanem łatania czyjejś nieudolności i lenistwa poprzez "idiotoodporność" softu. Lekki opierdol tańszy, szybszy i skuteczniejszy :P Ile tych umów wprowadza się dziennie, żeby to zrobić źle?

Reply:
A ja nie jestem fanem opierdalania za powtarzające się, sytuacje skoro można ich uniknąć w taki sposób - nie opierdol nie będzie skuteczniejszy - nie tędy droga. Pamiętaj, że to omyłka polegająca na niedokonaniu wyboru aniżeli źle dokonany wybór.

Reply:
Myślałem o tym, żeby pójść krok dalej i zrobić z tego selecta z wyborem np. formy działalności ale to już o wiele grubszy temat i na tą chwilę nie widzę PILNEJ potrzeby.

Reply:
Jeśli miałbym tak to robić to nie JS ale backend i pole automatycznie uzupełniać w `customeradd.php`.

Reply:
Wpis ma charakter humorystyczny. Przed przeczytaniem poluzuj krawat i miej pod ręką maść na ból tyłka.

Ironia ON

Nie. Nie ucz się chodzić. Nie ucz się czytać. Nie ucz się pisać. Życie zrobi wszystko za Ciebie. Przecież to takie oczywiste. Zwolnij wszystkich od razu u siebie, za chwilę LMS będzie sam wszystko robił. Ja już szykuję ziemniaki i olej. Zakład, że już szykujesz obsługę frytkownicy w LMS? ;) Będzie się aktywować na zmienną phpui.activate_new_fryer_device_because_i_think_that_lms_should_support_it. ;)

Wrzuć required od razu do każdego możliwego pola w całym LMS. Przecież za chwilę ktoś w Twoim biurze cię o to poprosi. A nie czekaj… nikt nie prosił. Sam tego chcesz ;) Bo ktoś jest dzbanem i nie potrafi zrobić tak prostej czynności, jaką jest wybranie jednego lub drugiego pola. Padłem.exe. Korygowanie czyjegoś niechlujstwa i błędów poprzez oprogramowanie. Pewnie w ten sam sposób narodził się autotune, a wraz z nim wytrysnęły wszystkie popowe pseudo wokalne gwiazdeczki.

Jeszcze moment i LMS uzyska samoświadomość i AI. Skynet nadchodzi! ;)

Ironia OFF

Reply:
Zdecydowanie masz rację - wszystkie pola potrzebne do prawidłowego uzupełnienia formularza powinny być oznaczone jako required. Dzięki temu nie trzeba potem programować obsługi błędu jest w efekcie mniej kodu i mniej żądań leci na serwer. Fajne to jest bo jak zapomnisz wyboru to po kliknięciu przycisku zapisz od razu przeglądarka Cię kieruje do tego pola.

To jest jak z walidacją PESELu i zdublowanych klientów. Można niewalidować z automatu - można kazać robić to ręcznie - a jeśli ktoś nie zrobi tego z pośpiechu to osctentacyjny opierdol najlepiej przy wszystkich innych pracownikach kręcić aferę z banana to wtedy efekt świetny haha ;-)

Ja tam uważam, że człowiek jest człowiekiem i ma prawo popełniać błędy ale trzeba skutecznie to utrudniać i walidować wstecz czy te błędy występują.

No ale mogę się mylić.

Reply:
> Ja tam uważam, że człowiek jest człowiekiem i ma prawo popełniać błędy ale trzeba skutecznie mu to utrudniać

zgadzam się z tym stwierdzeniem @interduo, zatem skoro opisane pomyłki użytkownika występują należy dążyć do ich zmniejszenia. Jednak w tym przypadku wprowadzenie 'requiered' na 'radiobutton' czy 'selectlisty' niczego nie wnosi, bo ostatecznie wybór musi być i tak dokonany świadomie przez użytkownika. Moim zdaniem lepiej by było np. ukrywać pola: REGON, KRS w przypadku zaznaczenia klienta jako osoby fizycznej. Wtedy użytkownik, wypełniając formularz, w chwili kiedy dojdzie do momentu wprowadzania REGON stwierdzi, że nie może tego zrobić i od razu cofnie się i przestawi typ osobowości prawnej. Oczywiście nie rozwiąże to wszystkich pomyłek, ale przemyślane ukrywanie/pokazywanie pól formularza może wpłynąć na ich zmniejszenie.

Reply:
@rafalpietraszewicz dobry pomysł na kolejne usprawnienie i uproszczenie wizualne formularza. 

Zwróć uwagę: 
1. uzupełnianie REGON i KRS nie jest wymagane nawet przy wyborze "firma",
2. w przypadku pola NIP nie możemy tak zrobić jak proponujesz - bo są osoby fizyczne, które chcą mieć NIP na FV, 

Mi chodzi o to by zaznaczane domyślnie wybory były tylko i wyłącznie dla mało istotnych pól lub pól. A to pole jest istotne bo wpływa na raport słany do GUS.

Reply:
> Oczywiście nie rozwiąże to wszystkich pomyłek, ale przemyślane ukrywanie/pokazywanie pól formularza może wpłynąć na ich zmniejszenie.

Warto rozważyć zamiast ukrywania poł ich disablowanie i przez to pokazowanie takim stylem, że od razu widać, że pole jest niedostępne. Zalety:
1. Brak pływania interfejsu.
2. Widać, że pole jest dostępne tylko zablokowane, więc trzeba coś zrobić, żeby odblokować.

Reply:
> Warto rozważyć zamiast ukrywania poł ich disablowanie i przez to pokazowanie takim stylem

Przy obecnym funkcjonowaniu formularza zdecydowanie jestem za 'disablowaniem'. To jest tylko kwestia odpowiedniego dobrania pól.

> 2\. w przypadku pola NIP nie możemy tak zrobić jak proponujesz - bo są osoby fizyczne, które chcą mieć NIP na FV

@interduo wiem o tym i zwróć uwagę że tego nie proponowałem :)

Generalnie trzeba zmierzać ku temu żeby było tak jak pisze @chilek: jeśli coś jest zablokowane, użytkownik musi coś zrobić (zaznaczyć prawidłowo opcję), żeby odblokować.

Reply:
Dobry pomysł @chilek
czyli do tego co zrobiłem by była pełnia szczęścia należało by dorobić:

Formularz customeradd:
jeśli dokonujemy wybór "osoba fizyczna" to blokujemy pola KRS, REGON,
jeśli dokonujemy wybór "firma" to blokujemy pole PESEL,

Formularz customeredit:
jeśli zmieniamy wybór na "osobę fizyczną" to wyłączyć po akceptacji formularza: czyścić pola KRS, REGON.
jeśli zmieniamy wybór na "firma" to wyłączyć i po akceptacji formularza: czyścić pole PESEL,

tak?

Reply:
OT BEGIN;
@interduo Ty napisałeś:
> Zwróć uwagę: w przypadku pola NIP nie możemy tak zrobić jak proponujesz - bo są osoby fizyczne, które chcą mieć NIP na FV,

Ja Ci mówię że tego nie proponowałem, a Ty mi odpisujesz:

> @rafalpietraszewicz tak zwróciłem uwagę, stąd nagłówek "Zwróć uwagę" zamiast "Przecież" ;-)

Niech się stanie. Niech Twoje zdanie będzie ostatnie :) Ja już do tego nie będę wracał. 
OT END;